### DISCUSSION TITLE
Utility meter and `sensor.energy_yield_total` - float not float(0)

### DISCUSSION THREAD

**Discussion by JensenNick** [JensenNick/huawei_solar_pees#11] [Forum]
> :exclamation: **If you have downloaded the *"Huawei Solar PEES package"* on or between Jan 22nd and Jan 26th 2024.**

In the e5b5d11435fb0f420506532979287dfa3a9e1c0b revision of the *"Huawei Solar PEES integration"* the default value of 0 (zero) was added to the input sensors for `sensor.energy_yield_total` along with other sensors, with the intention to make sure that it had a default value in setups with only one inverter. This approach however introduces a misbehavior of the utility meters using the `sensor.energy_yield_total`.

It is therefore important that float on `sensor.energy_yield_1` and `sensor.energy_yield_2` does NOT have a the default `float(0)` value in `sensor.energy_yield_total`. It should just be `float` and the sensor should have the availability check for both input sensors. This in kombination with a manually set state of `sensor.power_inverter_2` and `sensor.energy_inverter_2` to 0 (zero) according to the [README](https://github.com/JensenNick/huawei_solar_pees/blob/main/README.md#solar-pv-with-a-single-inverter-setup) will ensure correct behavior of all sensors and utility meters.

> :exclamation: **Here is the fix / correction needed. You can correct this manually in the `huawei_solar_pees.yaml` file in your package folder.**

![2024-01-26_23-15-54](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/94025020-3015-49bb-b4cf-fff49cc5bcb6)

![2024-01-26_22-42-47](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/a689cb2d-4eda-472c-8feb-e2cf7efe8c11)

**Comment by JacobsenKim** [comment]
> @JacobsenKim Purge og Reset Utility Meter inde under Developer Tools > SERVICES. Jeg er dog bange for at der er et problem med de float(0) jeg fik rettet den 22 januar. Se her https://github.com/JensenNick/huawei_solar_pees/issues/13

@JensenNick - Jeg flyttede den lige ind under float(0) tråden


Her er noget der bekræfter at en kilde til utility meter ikke må være 0 ved  ukendt.
https://community.home-assistant.io/t/utility-meter-and-source-sensor/551795

It’s because the utility meter ignores state changes from unavailable. This is a good thing or your utility meter would double every time the source sensor became unavailable. Not tracking while shutdown is a known side effect / limitation of this. Good news though. Looks like a fix for this will…

men der er så en ny (1 år gammel) setting til utility meter som måske løser problemet

https://github.com/home-assistant/core/pull/88446

This option is by default enabled so that it will not break the current implementation. This new feature could be turned on (i.e. set the option off) by either adding periodically_resetting: false to a configuration.yaml or change an existing utility meter through the options flow.

The last valid state is always set upon receiving a new valid state and restored on start of Home Assistant.

Jeg har selv været igang med samme project som dit her for ca. 1 år siden, nåede ikke så lang som du er, men kan se jeg har brugt denne funktion, mit project var baseret på sidste times forbrug kWh og sidste times pris for at få stabilitet i tallene, men jeg støtte på for mange problemer jeg ikke kunne løse, samt jeg kunne se at jeg ville ende med 100 vis af sensor så projektet blev henlagt.

**Reply by JensenNick** [reply]
@JacobsenKim Tak for input. Jeg har endnu ikke fået sat en testmaskine op (og det bliver jeg straffet for lige nu :-) ) og umiddelbart lyder det jo fornuftigt med `periodically resetting: false`, men jeg er nød til at få fixet de problemer jeg har givet mig selv og herefter tænker jeg at sådan noget som dette kan testes på en maskine hertil. Har du en fornemmelse om det var denne funktion der gav problemer i dit setup?

**Reply by JacobsenKim** [reply]
> Har du en fornemmelse om det var denne funktion der gav problemer i dit setup?

Nej, det var for meget at gå i gang med den gang, også oplevede jeg problemer med apex chart når HA blev genstartet så flippede de ud, men fandt ikke løsningen på det.

Men det er nemt at teste.
Lav 1 ny sensor ud fra en sensor du ved dumper og brug float(0), og lav så nogle utility meter til sensor 1 med periodically resetting: false og 1 uden.
Du kan kopiere nogle fra den kode du har også om døbe sensor også have dem i en test.yaml i din private pakke som du kan teste med, så påvirker din test ikke din packet.

også drop float(0) til det er gennem testet med periodically resetting: false

**Reply by JensenNick** [reply]
@JacobsenKim Yes. Har mange bolde i luften and andre projekter der skal udvikles og søsættes, så det får ikke 1. prioritet hos mig :smiley:.

**Reply by JacobsenKim** [reply]
> Jeg har endnu ikke fået sat en testmaskine op (og det bliver jeg straffet for lige nu :-) )

Det er ikke nogen fordel at have en test maskine med mindre man slukker sin live maskine, da inverter kun håndtere 1 HA af gangen.

Jeg har kørt en VM fra i går og til i dag, billedet fra min live maskine taler for sig selv. 

![Skærmbillede fra 2024-01-29 14-28-12](https://github.com/JensenNick/huawei_solar_pees/assets/157890151/4f1d5c13-7211-468f-b507-9978c15ec103)
